src/main/scala/com/redislabs/provider/redis/streaming/RedisInputDStream.scala

```diff
@@ -0,0 +1,74 @@
+package com.redislabs.provider.redis.streaming
+
+import com.redislabs.provider.redis.RedisConfig
+import org.apache.curator.utils.ThreadUtils
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.streaming.StreamingContext
+import org.apache.spark.streaming.receiver.Receiver
+import org.apache.spark.streaming.dstream.ReceiverInputDStream
+
+import redis.clients.jedis._
+
+import scala.reflect.{ClassTag, classTag}
+import scala.util.control.NonFatal
+
+class RedisInputDStream[T: ClassTag](_ssc: StreamingContext,
+                                     keys: Array[String],
+                                     storageLevel: StorageLevel,
+                                     redisConfig: RedisConfig,
+                                     streamType: Class[T])
+  extends ReceiverInputDStream[T](_ssc) {
+  def getReceiver(): Receiver[T] = {
+    new RedisReceiver(keys, storageLevel, redisConfig, streamType)
+  }
+}
+
+
+private class RedisReceiver[T: ClassTag](keys: Array[String],
+                                         storageLevel: StorageLevel,
+                                         redisConfig: RedisConfig,
+                                         streamType: Class[T])
+  extends Receiver[T](storageLevel) {
+
+  def onStart() {
+    val executorPool = ThreadUtils.newFixedThreadPool(keys.length, "BlockLists Streaming")
```

Comment:
ok, makes sense. I'm wondering though what will happen with many many keys. but leave it like this for now